src/plugins/presentation_util/public/components/field_picker/field_picker.tsx

```diff
@@ -31,115 +32,94 @@ export const FieldPicker = ({
-          )
+  const availableFields = useMemo(
+    () =>
+      sortBy(
+        (dataView?.fields ?? [])
+          .filter((f) => typesFilter.length === 0 || typesFilter.includes(f.type as string))
           .filter((f) => (filterPredicate ? filterPredicate(f) : true)),
         ['name']
-      )
-    : [];
+      ),
+    [dataView, filterPredicate, typesFilter]
+  );
+
+  useEffect(() => {
+    if (!dataView) return;
+    const options: EuiSelectableOption[] = (availableFields ?? []).map((field) => {
+      return {
+        key: field.name,
+        label: field.displayName ?? field.name,
+        className: classNames('presFieldPicker__fieldButton', {
+          presFieldPickerFieldButtonActive: field.name === selectedFieldName,
+        }),
+        'data-test-subj': `field-picker-select-${field.name}`,
+        prepend: <FieldIcon type={field.type} label={field.name} scripted={field.scripted} />,
+      };
+    });
+    setFieldSelectableOptions(options);
+  }, [availableFields, dataView, filterPredicate, selectedFieldName, typesFilter]);
 
-  const uniqueTypes = dataView
-    ? uniq(
-        dataView.fields
-          .filter((f) => (filterPredicate ? filterPredicate(f) : true))
-          .map((f) => f.type as string)
-      )
-    : [];
+  const uniqueTypes = useMemo(
+    () =>
+      dataView
+        ? uniq(
+            dataView.fields
+              .filter((f) => (filterPredicate ? filterPredicate(f) : true))
+              .map((f) => f.type as string)
+          )
+        : [],
+    [dataView, filterPredicate]
+  );
+
+  const fieldTypeFilter = (
+    <FieldTypeFilter
+      onFieldTypesChange={(types) => setTypesFilter(types)}
+      fieldTypesValue={typesFilter}
+      availableFieldTypes={uniqueTypes}
+    />
+  );
 
   return (
-    <EuiFlexGroup
-      direction="column"
-      alignItems="stretch"
-      gutterSize="s"
-      className={`presFieldPicker__container ${
-        !dataView && 'presFieldPicker__container--disabled'
-      }`}
+    <EuiSelectable
+      emptyMessage={i18n.translate('presentationUtil.fieldPicker.noFieldsLabel', {
+        defaultMessage: 'No matching fields',
+      })}
+      aria-label={i18n.translate('presentationUtil.fieldPicker.selectableAriaLabel', {
+        defaultMessage: 'Select a field',
+      })}
+      searchable
+      options={fieldSelectableOptions}
+      onChange={(options, _, changedOption) => {
+        setFieldSelectableOptions(options);
+        if (!dataView || !changedOption.key) return;
+        const field = dataView.getFieldByName(changedOption.key);
+        if (field) onSelectField?.(field);
+      }}
+      searchProps={{
+        'data-test-subj': 'field-search-input',
+        placeholder: i18n.translate('presentationUtil.fieldSearch.searchPlaceHolder', {
+          defaultMessage: 'Search field names',
+        }),
+      }}
+      listProps={{
+        isVirtualized: true,
+        showIcons: false,
```

Comment:
Since there is often a scrollbar, I think that `bordered: true` looks a bit nicer?


![EuiSelectableBorder](https://user-images.githubusercontent.com/8698078/221034816-5b07f401-6597-4d83-8f8d-3eb9e51c4aaf.png)

Reply:
+1 on Hannah's suggestion, a border would be a nice addition

Reply:
I felt that the border looked a bit too noisy, but seeing the screenshots (and knowing that y'all are down for the border), I'll add it!